File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdbool.h>
#include <math.h>

int len_num(int num) {
    if (num == 0)
        return 1;
    int count = 0;
    while (num) {
        num /= 10;
        count++;
    }
    return count;
}

bool is_armstrong_number(int candidate) {
    int original = candidate;
    int digit_num = len_num(candidate);
    int total = 0;
    while (candidate) {
        int digit = candidate % 10;
        total += pow(digit, digit_num);
        candidate /= 10;
    }
    return (original == total);
}

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
import math

def len_num(num):
    if num == 0:
        return 1
    return math.floor(math.log10(abs(num))) + 1

def is_armstrong_number(number):
    original = number
    digit_num = len_num(number)
    total = 0

    while number:
        digit = number % 10
        total += digit ** digit_num
        number //= 10
    
    return original == total